multi-agent/lib/mas/elements/sandboxes/common/sandbox_factories.py

```
@@ -0,0 +1,296 @@
+    body: bytes,
+    session_id: Optional[str] = None,
+) -> tuple:
+    """Send a POST to the MCP endpoint, return (response_body, response_headers)."""
+    req_headers = {
+        **headers,
+        "Content-Type": "application/json",
+        "Accept": "text/event-stream, application/json",
+    }
+    if session_id:
+        req_headers["Mcp-Session-Id"] = session_id
+    req = urllib.request.Request(url, method="POST", headers=req_headers, data=body)
```

Ruff (0.15.20)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)